snapshot-labs/snapshot
# feat: Use sidekick server to handle CSV report download

### Issues

Fixes #3756

Use https://github.com/snapshot-labs/snapshot-sidekick to handle the votes CSV report download

### Changes 

1. Refactor the `useReportDownload` composable to download the complete file from sidekick server, instead of building it from snapshot-hub requests.
2. Show error message in case of download errors

### How to test

1. Go to a closed proposal
3. Try to download the votes CSV report

- It should download the CSV file if the file is cached on sidekick
- It should show a modal with a pending generation error message if the file is not ready yet on sidekick
- It should show a modal with a generic error message on all other errors (sidekick not online) 

### To-Do

- [x] Waiting for sidekick server to be online/deployed 

### Self-review checklist

This PR is ready, waiting to backend server to be online before merging
